src/script/lua_api/l_metadata.cpp

```diff
@@ -289,8 +296,14 @@ bool MetaDataRef::handleFromTable(lua_State *L, int table, IMetadata *meta)
 		while (lua_next(L, fieldstable) != 0) {
 			// key at index -2 and value at index -1
 			std::string name = readParam<std::string>(L, -2);
-			auto value = readParam<std::string_view>(L, -1);
-			meta->setString(name, value);
+			if (lua_type(L, -2) == LUA_TNUMBER) {
+				// If someone accidentally gives us a number,
+				// we shouldn't silently lose data by letting Lua convert it to string.
```

Comment:
should we?
I see this becoming yet another annoying implementation detail that some mod inevitably relies on in a few years.
`from_table` is for passing `to_table`'d stuff in there. not a bulk alternative to `set_<type>`.

Reply:
okay, but then there should probably be a deprecation warning for non-string values, or at least a warning if precision is lost?

(in general i think we'll need to be more strict about checking types at runtime.)

Reply:
sure.

Reply:
okay